app/uk/gov/hmrc/helptosave/util/Logging.scala

```diff
@@ -29,42 +32,52 @@ object Logging {
 
   implicit class LoggerOps(val logger: Logger) {
 
-    def debug(message: String, nino: NINO)(implicit transformer: NINOLogMessageTransformer): Unit =
-      logger.debug(transformer.transform(message, nino))
+    def debug(message: String, nino: NINO)(implicit transformer: LogMessageTransformer, hc: HeaderCarrier): Unit =
+      logger.debug(transformer.transform(message, nino, getCorrelationId()))
 
-    def info(message: String, nino: NINO)(implicit transformer: NINOLogMessageTransformer): Unit =
-      logger.info(transformer.transform(message, nino))
+    def info(message: String, nino: NINO)(implicit transformer: LogMessageTransformer, hc: HeaderCarrier): Unit =
+      logger.info(transformer.transform(message, nino, getCorrelationId()))
 
-    def warn(message: String, nino: NINO)(implicit transformer: NINOLogMessageTransformer): Unit =
-      logger.warn(transformer.transform(message, nino))
+    def warn(message: String, nino: NINO)(implicit transformer: LogMessageTransformer, hc: HeaderCarrier): Unit =
+      logger.warn(transformer.transform(message, nino, getCorrelationId()))
 
-    def warn(message: String, e: ⇒ Throwable, nino: NINO)(implicit transformer: NINOLogMessageTransformer): Unit =
-      logger.warn(transformer.transform(message, nino), e)
+    def warn(message: String, e: ⇒ Throwable, nino: NINO)(implicit transformer: LogMessageTransformer, hc: HeaderCarrier): Unit =
+      logger.warn(transformer.transform(message, nino, getCorrelationId()), e)
 
-    def error(message: String, nino: NINO)(implicit transformer: NINOLogMessageTransformer): Unit =
-      logger.error(transformer.transform(message, nino))
+    def error(message: String, nino: NINO)(implicit transformer: LogMessageTransformer, hc: HeaderCarrier): Unit =
+      logger.error(transformer.transform(message, nino, getCorrelationId()))
 
-    def error(message: String, e: ⇒ Throwable, nino: NINO)(implicit transformer: NINOLogMessageTransformer): Unit =
-      logger.error(transformer.transform(message, nino), e)
+    def error(message: String, e: ⇒ Throwable, nino: NINO)(implicit transformer: LogMessageTransformer, hc: HeaderCarrier): Unit =
+      logger.error(transformer.transform(message, nino, getCorrelationId()), e)
 
+    def getCorrelationId()(implicit hc: HeaderCarrier): Option[String] =
```

Comment:
`getCorrelationId` doesn't belong in the implicit class for `Logger` - it doesn't extend the behaviour of `Logger`, it extends the behaviour of `HeaderCarrier`. It would make more sense have a private method outside of this implicit class to get the behaviour you want